core/src/main/java/de/bwaldvogel/mongo/backend/AbstractMongoCollection.java

```diff
@@ -436,15 +439,28 @@ public synchronized Document findAndModify(Document query) {
             orderBy = null;
         }
 
-        Iterable<Document> objs = queryDocuments(query, orderBy, numberToSkip, numberToReturn);
+        QueryResult<Document> objs = queryDocuments(query, orderBy, numberToSkip, numberToReturn);
 
         if (fieldSelector != null && !fieldSelector.keySet().isEmpty()) {
-            return new ProjectingIterable(objs, fieldSelector, idField);
+            return new QueryResult<>(new ProjectingIterable(objs, fieldSelector, idField), 0);
         }
 
         return objs;
     }
 
+    @Override
+    public synchronized QueryResult<Document> handleGetMore(long cursorId, int numberToReturn) {
+        if (!cursors.containsKey(cursorId)) {
+            throw new RuntimeException("Cursor not found");
+        }
+        Cursor cursor = cursors.get(cursorId);
+        List<Document> docs = new ArrayList<>();
+        while (cursor.getDocuments().size() > 0 && docs.size() < numberToReturn) {
+            docs.add(cursor.getDocuments().poll());
+        }
+        return new QueryResult<>(docs, cursor.isEmpty() ? 0 : cursorId);
```

Comment:
When do you remove cursor from `AbstractMongoBackend.cursors` map ? 
Is there a cleanup happening ?

Reply:
Right now I'm not implementing any clean up. I left that open, see MongoDatabaseHandler line 88

I would say this is low priority since there should be a clean up after every test.

Reply:
Couldn’t we easily remove a cursor once no documents are remaining?

Reply:
Addressd on commit 0f4cc05